status/versions.go

```diff
@@ -312,18 +320,52 @@ func kubernetesVersion() (*ExternalServiceInfo, error) {
 // did that that version change will matter, and the kiali pod could be bounced as a workaround.
 var istioSupportsCanonical *bool
 
-// IstioSupportsCanonical returns true if Istio version can be determined and is >= 1.5.
-// TODO: This test can be removed when Kiali stops supporting any Istio versions < 1.5
+// IstioSupportsCanonical returns true if Telemetry V2 is enabled
+// TODO: This test can be removed when Kiali stops supporting Istio versions with Mixer Telemetry
 func IstioSupportsCanonical() bool {
 	if istioSupportsCanonical != nil {
 		return *istioSupportsCanonical
 	}
 
-	istioVersion, err := istioVersion()
+	k8sConfig, err := kubernetes.ConfigClient()
+	if err != nil {
+		log.Warningf("IstioSupportsCanonical: Cannot create config structure Kubernetes Client.")
+		return false
+	}
+
+	k8s, err := kube.NewForConfig(k8sConfig)
+	if err != nil {
+		log.Warningf("IstioSupportsCanonical: Cannot create Kubernetes Client.")
+		return false
+	}
+
+	cfg := config.Get()
+	istioConfig, err := k8s.CoreV1().ConfigMaps(cfg.IstioNamespace).Get(ISTIO_CONFIGMAP_NAME, meta_v1.GetOptions{})
```

Comment:
Does it make sense to move this function into the KubernetesClient instead? This way all calls to k8s are group into the same client. Plus, It also helps testing this feature.
Maybe adding a GetConfigMap(name, namespace string).
wdyt?

Reply:
I moved to the k8s client :)